cmd/minikube/cmd/start.go

```diff
@@ -266,6 +293,60 @@ func showKubectlConnectInfo(kubeconfig *pkgutil.KubeConfigSetup) {
 	}
 }
 
+func selectImageRepository(config cfg.Config) (bool, string, error) {
+	mirrorCountry := strings.ToLower(viper.GetString(imageMirrorCountry))
+	repos := constants.ImageRepositories
+	var countries []string
+	if mirrorCountry != "" {
+		_, ok := repos[mirrorCountry]
+		if !ok {
+			return false, "", fmt.Errorf("invalid image mirror country code: %s", mirrorCountry)
+		}
+		countries = []string{mirrorCountry, ""}
+
+	} else {
+		// make sure global is preferred
+		countries = []string{""}
+		for k := range repos {
```

Comment:
I really like this, but I don't feel we should implement alternative mirrors by default until we have a way of validating the contents of the images (#4104). My concern is the case where a business blocks gcr.io and other known image repositories, and we fall back to a 3rd party poisoned repository for them, and deliver them a backdoored image.

In the short term, I don't mind if we have country fallbacks if we have a reasonable suspicion that the user is in that country - but I hesitate to apply it globally until #4104 can be fixed.

Reply:
This logic is only triggered manually by setting --image-repository to auto or --image-mirror-country is set before #4104 is fixed.